modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM678ApplicationCreationTestCase.java

```diff
@@ -108,262 +106,146 @@ public void setEnvironment() throws Exception {
 
     //create application with valid data
-    @Test(groups = "webapp", dataProvider = "createApplicationWithValidData", description = "Create an Application")
+    @Test(dataProvider = "createApplicationWithValidData", description = "Create an Application")
     public void testApplicationCreation(String applicationName, String tier, String description)
             throws Exception {
 
-        HttpResponse addApplicationResponse = apiStore.addApplication(applicationName, appTier, "", description);
-        assertEquals(addApplicationResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Response Code is mismatched in add application: " + applicationName);
-        JSONObject addApplicationJsonObject = new JSONObject(addApplicationResponse.getData());
-        assertFalse(addApplicationJsonObject.getBoolean("error"), "Error in Application Creation: " + applicationName);
-        assertEquals(addApplicationJsonObject.get("status"), "APPROVED", "Error in Application Creation: "
-                + applicationName);
 
+        HttpResponse applicationResponse = restAPIStore.createApplication(applicationName,
+                description, appTier,
+                ApplicationDTO.TokenTypeEnum.OAUTH);
+        applicationId = applicationResponse.getData();
+        assertEquals(applicationResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                "Response Code is mismatched in add application " + applicationName);
+        assertNotNull(applicationResponse, "Error in Application Creation: "
+                + applicationName);
         //add applications to a list (for get all applications and deletion purpose when test finished)
         applicationsList.add(applicationName);
     }
 
+    //TODO: Commented until fix: https://github.com/wso2/product-apim/issues/6012
+
     //Create Application with invalid data
-    @Test(groups = "webapp", dataProvider = "createApplicationWithInValidData", description = "Create application for invalid data")
-    public void testApplicationCreationForInvalidData(String applicationName, String tier, String description)
-            throws Exception {
-        HttpResponse addApplicationResponse = apiStore.addApplication(applicationName, appTier, "", description);
-        JSONObject addApplicationJsonObject = new JSONObject(addApplicationResponse.getData());
-        //Error when application name string empty (JIRA 4040)
-        //assertTrue(addApplicationJsonObject.getBoolean("error"), "Error in Application Creation with Invalid data: "
-        // + applicationName);
-
-        //Verify application is exists
-        HttpResponse getAllApplicationsResponse = apiStore.getAllApplications();
-        JSONObject allApplicationsJsonObject = new JSONObject(getAllApplicationsResponse.getData());
-//        assertFalse(allApplicationsJsonObject.getBoolean("error"), "Error in App verification");
-    }
+//    @Test(groups = "webapp", dataProvider = "createApplicationWithInValidData",
```

Comment:
Do we need this commented lines?

Reply:
Yes. its comment until fix: https://github.com/wso2/product-apim/issues/6012